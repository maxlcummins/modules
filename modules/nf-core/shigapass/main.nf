process SHIGAPASS {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/shigapass:A1.5.0--hdfd78af_0':
        'biocontainers/shigapass:1.5.0--hdfd78af_0' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("${prefix}.csv"), emit: report
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def is_compressed = fasta.getName().endsWith(".gz") ? true : false
    def fasta_name = fasta.getName().replace(".gz", "")
    """
    # ShigaPass does not accept compressed FASTA files
    if [ "$is_compressed" == "true" ]; then
        gzip -c -d $fasta > $fasta_name
    fi

    # Convert our genome path to a file with a path in it
    ls $fasta_name > ${fasta_name}_tmp.txt

    # Run ShigaPass
    ShigaPass.sh \\
        -l ${fasta_name}_tmp.txt \\
        $options.args \\
        -p /usr/local/share/shigapass-1.5.0/db \\
        -t $task.cpus \\
        -o ${prefix}

    # Remove the temporary file from above
    rm ${fasta_name}_tmp.txt

    # Delete the directory generated by ShigaPass
    mv ${prefix}/*.csv ${prefix}.csv

    rmdir ${prefix}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        shigapass: \$(echo \$(ShigaPass.sh --version 2>&1) | sed 's/^.*ShigaPass version //' )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.csv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        shigapass: \$(echo \$(ShigaPass.sh --version 2>&1) | sed 's/^.*ShigaPass version //' )
    END_VERSIONS
    """
}
