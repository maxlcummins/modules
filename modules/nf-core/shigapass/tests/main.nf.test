nextflow_process {

    name "Test Process SHIGAPASS"
    script "../main.nf"
    process "SHIGAPASS"
    tag "modules"
    tag "modules_nfcore"
    tag "shigapass"

    test("candidatus_portiera_aleyrodidarum - genome.fasta") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                                [ id:'test', single_end:false ], // meta map
                                file(params.modules_testdata_base_path + '/genomics/prokaryotes/candidatus_portiera_aleyrodidarum/genome/genome.fasta', checkIfExists: true)
                            ]
                input[1] = []
                """
            }
        }

        then {
            assertAll(
            { assert process.success },
            { assert snapshot(process.out).match() }
            )
        }
    }

    test("candidatus_portiera_aleyrodidarum - genome.fasta - stub") {

        options "-stub"

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                                [ id:'test', single_end:false ], // meta map
                                file(params.modules_testdata_base_path + '/genomics/prokaryotes/candidatus_portiera_aleyrodidarum/genome/genome.fasta', checkIfExists: true)
                            ]
                input[1] = []
                """
            }
        }

        then {
            assertAll(
            { assert process.success },
            { assert snapshot(process.out).match() }
            )
        }

    }

}
