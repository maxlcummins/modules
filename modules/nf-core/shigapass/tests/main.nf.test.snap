{
    "candidatus_portiera_aleyrodidarum - genome.fasta - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "report": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-13T11:41:51.692438"
    },
    "candidatus_portiera_aleyrodidarum - genome.fasta": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "report": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-13T11:41:48.121725"
    }
}